pkg/controllers/user/helm/common/common.go

```diff
@@ -143,6 +139,52 @@ func InstallCharts(rootDir, port string, obj *v3.App) error {
 	return nil
 }
 
+func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
+	setValues := []string{}
+	// a user-supplied values file will overridden default values.yaml
+	if app.Spec.ValuesYaml != "" {
+		valuesYaml := filepath.Join(tempDir, "answers.yaml")
+		if err := ioutil.WriteFile(valuesYaml, []byte(app.Spec.ValuesYaml), 0755); err != nil {
+			return setValues, err
+		}
+		setValues = append(setValues, "--values", valuesYaml)
+	}
+
+	// `--set` values will overridden the user-supplied values.yaml file
+	if app.Spec.Answers != nil {
+		answers := app.Spec.Answers
+		var values, strValues []string
+		for k, v := range answers {
+			strValue := rancherTypedVal(k, v)
+			if len(strValue) > 0 {
+				strValues = append(strValues, fmt.Sprintf("%s=%s", k, strValue))
+			} else {
+				values = append(values, fmt.Sprintf("%s=%s", k, v))
+			}
+		}
+		setValues = append(setValues, "--set", strings.Join(values, ","))
+		setValues = append(setValues, "--set-string", strings.Join(strValues, ","))
+	}
+	return setValues, nil
+}
+
+// rancherTypedVal parse value based on rancher type prefix
+func rancherTypedVal(key, value string) string {
+	parts := strings.SplitN(value, ":", 2)
+	if len(parts) == 2 {
+		t := parts[0]
+		switch t {
+		// we only need to deal the string: prefix for now(e.g string:11111 will be converted to string `11111`),
+		// for the other types like bool, int will be passed by default `--set`
+		case "string":
```

Comment:
is the prefix "string:" special enough? Or would we need to make it more special like helm-string or rancher-string

Reply:
dont change yet, ill discuss with @vincent99

Reply:
helm uses syntax like `!!str 123` to specify string value in the values.yaml, but it won't work in the `--set`, but we may still use the same syntax, and make it works in the `--set` like what we did in here.

Reply:
https://github.com/helm/helm/blob/master/docs/chart_template_guide/yaml_techniques.md#scalar-types-in-yaml